pkg/process_manager.go

```diff
@@ -31,19 +32,151 @@ type ProcessStatus struct {
 	State    ProcessState
 }
 
-type processStream struct {
-	// TODO
+// sharedBufWriter is the write half of a shared buffer that allows multiple
+// readers.
+type sharedBufWriter struct {
+	bufLock sync.RWMutex
+	buf     []byte
+
+	closed atomic.Bool
+
+	readersLock sync.RWMutex
```

Comment:
I think a regular `sync.Mutex` might fit better here since the single writer uses read locks and the multiple readers use regular locks.  In practice there shouldn't ever be multiple read locks on a single `sharedBufWriter` unless the `Close()` method were called mid write, which I don't think warrants the extra overhead.

Reply:
Ah, yeah, good point.